apps/api/src/app/workflows-v2/usecases/extract-variables/extract-variables.usecase.ts

```diff
@@ -68,71 +71,20 @@ export class BuildPayloadSchema {
     for (const [key, value] of Object.entries(controlValue)) {
       if (isStringTipTapNode(value)) {
         processedValue[key] = this.hydrateEmailSchemaUseCase.execute({ emailEditor: value });
+      } else if (key === 'skip' && isValidRule(value as RulesLogic<AdditionalOperation>)) {
+        const fields = extractFieldsFromRules(value as RulesLogic<AdditionalOperation>)
+          .filter((field) => field.startsWith('payload.') || field.startsWith('subscriber.data.'))
```

Comment:
How will previous steps be handled?

Reply:
The variables for the previous steps are built in `BuildVariableSchemaUsecase`. 
The `ExtractVariables` usecase extracts all the variables from the control values, for example, from the skip rules (skip conditions).